pkg/connection/store/kubernetes/store_test.go

```diff
@@ -118,6 +118,22 @@ func TestSecretStoreReadKeyValues(t *testing.T) {
 				result: store.KeyValues(fakeKV()),
 			},
 		},
+		"SecretNotFound": {
+			reason: "Should return an empty secret and without any error if the secret is not found",
+			args: args{
+				client: resource.ClientApplicator{
+					Client: &test.MockClient{
+						MockGet: test.NewMockGetFn(kerrors.NewNotFound(schema.GroupResource{}, "")),
+					},
+				},
+				n: store.ScopedName{
+					Name: fakeSecretName,
+				},
+			},
+			want: want{
+				result: nil,
```

Comment:
if the test name says that `return an empty secret`, should then the result be an empty map? If it should remain `nil`, then perhaps we should rephrase the reason above to reflect that?

Reply:
What we `want` is not the secret, but the error. Therefore, the test is correct

Reply:
> What we `want` is not the secret, but the error. Therefore, the test is correct

Nevermind, that I thought is not the one I wrote. I'll update it.

Reply:
Not sure that I follow: in this test `ReadKeyValues` is not going to return the error, and then we compare secrets data with `want.result`. Do we consider an empty secret the one with `Data` field being set to `nil` or to an empty map?

Reply:
The zero value of a map is nil. I observed different behaviors like while printing it, I get `map[]` but in the `cmp.Diff` method, it considers it as a `nil`.